src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisState.PerAnalyzerState.cs

```diff
@@ -375,9 +397,9 @@ public bool TryStartProcessingEvent(CompilationEvent compilationEvent, [NotNullW
                 return TryStartProcessingEntity(compilationEvent, _pendingEvents, _analyzerStateDataPool, out state);
             }
 
-            public void MarkEventComplete(CompilationEvent compilationEvent)
+            public bool TryMarkEventComplete(CompilationEvent compilationEvent)
```

Comment:
I converted all the methods that mark completion of events/symbols/declaration/syntax from `void MarkXXXComplete(...)` to `bool TryMarkXXXComplete(...)`